experimental/dds/tree2/src/shared-tree-core/branch.ts

```diff
@@ -42,17 +42,17 @@ import { TransactionStack } from "./transactionStack";
  * transaction completes and all pending commits are replaced with a single squash commit.
  */
 export type SharedTreeBranchChange<TChange> =
-	| { type: "append"; change: TaggedChange<TChange>; newCommits: GraphCommit<TChange>[] }
+	| { type: "append"; change: TaggedChange<TChange>; newCommits: readonly GraphCommit<TChange>[] }
```

Comment:
These readonlies make it easier to type the new `changeEvent` constants below and should have been readonly in the first place.